games/devtest/mods/testtranslations/translation_mo.fr.po

```diff
@@ -0,0 +1,9 @@
+msgctxt "testtranslations"
+msgid "Testing .mo files: untranslated"
+msgstr "Testing .mo files: translated"
+
+msgctxt "testtranslations"
```

Comment:
If this msgctxt is required for every message, this should be checked by the engine and warned/errored about

Reply:
I assume this file is used to generate `locale/translation_mo.fr.mo`? The translation file reader uses the base filename (here: `translation_mo`) as the text domain if one is not specified.

The [GNU Gettext doc on PO files](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html) does not seem clear about about the scope (i.e. the entries that the `msgctxt` applies to) of `msgctxt`. However, considering that
* The documentation states that "an empty context string and an absent msgctxt line do not mean the same thing";
* Entries with `msgctxt` are considered as "entries with a context specifier"; and
* The entire documentation does not mention "scopes" of other specifiers (that said, having `msgid` and `msgtxt` to multiple entries would make little sense anyway)

It would be safe to assume that entries without an explicit context specifier would be considered as entries without context information, which, combined with the fact that Minetest does not differentiate between "text domain" and "context" (and, in this PR, uses the former if the latter is absent), means that any entry without an explicit `msgctxt` specifier would be put under the default (here: `translation_mo`) domain, which would explain the duplicate `msgctxt`.

It is probably also worth noting that the generated `.mo` file includes context information on every entry as well.

On a less related note, while entries without `msgctxt` are allowed (see https://github.com/minetest/minetest/pull/13687#discussion_r1518407844), there does not seem to be any testcase/testfile related to this.